anndata/core/aligned_mapping.py

```diff
@@ -45,15 +43,32 @@ class AlignedMapping(cabc.MutableMapping, ABC):
     to either one or both AnnData axes.
     """
 
+    _axis: Optional[int]
+    _parent: Union["anndata.AnnData", "raw.Raw"]
+
     _allow_df: ClassVar[bool]
     """If this mapping supports heterogeneous DataFrames"""
 
-    _view_class: ClassVar[Type["AlignedViewMixin"]]
+    _View: ClassVar[Type["AlignedView"]]
     """The view class for this aligned mapping."""
 
-    _actual_class: ClassVar[Type["AlignedActualMixin"]]
+    _Actual: ClassVar[Type["AlignedActual"]]
     """The actual class (which has it's own data) for this aligned mapping."""
 
+    def __new__(
+        cls,
+        parent: Union["anndata.AnnData", "raw.Raw"],
+        aligned_to: Union[Optional[int], "AlignedMapping"] = None,
+        *,
+        subset_idx: Optional[OneDIdx] = None,
+        vals: Mapping[str, V] = MappingProxyType({}),
+    ):
+        if isinstance(aligned_to, (int, type(None))):
+            return super().__new__(cls._Actual)
+        else:  # Initialize view
+            assert subset_idx is not None, "A view needs a subset_idx"
+            return super().__new__(cls._View)
+
```

Comment:
As mentioned in https://github.com/theislab/anndata/pull/260/files/47a688c0dbf59c488eaec40a30f1d462f6d75377#r352968031, I think this should be reverted.

Reply:
answered there